libs/server/Transaction/TransactionManager.cs

```diff
@@ -227,8 +227,14 @@ internal bool RunTransactionProc(byte id, ref CustomProcedureInput procInput, Cu
             {
                 try
                 {
-                    // Run finalize procedure at the end
-                    proc.Finalize(garnetTxFinalizeApi, ref procInput, ref output);
+                    // Run finalize procedure at the end.
+                    // If the transaction was invoked during AOF replay skip the finalize step altogether
+                    // Finalize logs to AOF accordingly, so let the replay pick up the commits from AOF as
+                    // part of normal AOF replay.
```

Comment:
Put a comment in the transaction proc base class/interface that Finalize is considered post transaction processing and therefore is not executed at recovery time. Instead, the individual Tsavorite commands are logged and replayed through the AOF.